addons/website_sale_stock/static/src/js/website_sale.js

```diff
@@ -0,0 +1,92 @@
+/** @odoo-module alias=website_sale_stock.website_sale**/
+
+import {WebsiteSale} from 'website_sale.website_sale';
+import VariantMixin from "sale.VariantMixin";
+import wSaleUtils from 'website_sale.utils';
+import {is_email} from 'web.utils';
+
+WebsiteSale.include({
+
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        'click #add_stock_email_notification_product_message': '_onClickAddStockEmailNotificationProductMessage',
+        'click #add_stock_email_notification_product_form_button': '_onSubmitAddStockEmailNotificationProductForm',
+    }),
+
+    _onClickAddStockEmailNotificationProductMessage: function (ev) {
+        ev.currentTarget.classList.add('d-none');
+        const partner_email = document.querySelector('#partner_email').value;
+        const form = ev.currentTarget.parentElement.querySelector('#add_stock_email_notification_product_form');
+        const email_input = form.querySelector('input[name="email"]');
+
+        email_input.value = partner_email;
+        form.classList.remove('d-none');
+
+    },
+
+    _onSubmitAddStockEmailNotificationProductForm: function (ev) {
+        const self = this;
+        const form = ev.currentTarget.closest('#add_stock_email_notification_product_form');
+        const product_id = parseInt(form.querySelector('input[name="product_id"]').value);
+        const email = form.querySelector('input[name="email"]').value.trim();
+        const incorrect_icon = form.querySelector('#add_stock_email_notification_product_input_incorrect');
+        if (!is_email(email)) {
+            incorrect_icon.classList.remove('d-none');
+            return
+        }
+
+        this._rpc({
+            route: "/shop/add_stock_email_notification",
+            params: {
+                product_id,
+                email
+            },
+        }).then(function (data) {
+            if (data === '400 Bad Request: Invalid Email'){
+                incorrect_icon.classList.remove('d-none');
+                return
+            }
+            const div = form.closest('#add_stock_email_notification_product_div');
+            const message = div.querySelector('#add_stock_email_notification_product_success_message');
+
+            message.classList.remove('d-none');
+            form.classList.add('d-none');
+
+            // This is to add the product to the wishlist since it creates one.
+            // We have to check that website_sale_wishlist is installed to add it.
```

Comment:
<img width="962" alt="Screenshot 2022-05-31 at 10 44 54" src="https://user-images.githubusercontent.com/101560055/171132114-b4aac9bc-99af-4650-93d5-4e82e0e75ca3.png">

This is to trigger the animation that adds a product in the wishlist so I mean that `website_stock` could be installed and not `website_sale_wishlist`, in which case `self.wishlistProductIDs` wouldn't exists since it is added by `website_sale_wishlist`.

Another idea would be to do that in website_sale_stock_wishlist by adding behaviour to the `_onSubmitAddStockEmailNotificationProductForm` function since website_sale_stock_wishlist depends on `website_sale_wishlist` and `website_sale_wishlist` ? What do you think ?